src/transformers/modeling_outputs.py

```diff
@@ -841,3 +882,31 @@ class SemanticSegmentationModelOutput(ModelOutput):
     logits: torch.FloatTensor = None
     hidden_states: Optional[Tuple[torch.FloatTensor]] = None
     attentions: Optional[Tuple[torch.FloatTensor]] = None
+
+
+@dataclass
+class ImageClassificationModelOutput(ModelOutput):
+    """
+    Base class for outputs of image classification models.
+
+    Args:
+        loss (`torch.FloatTensor` of shape `(1,)`, *optional*, returned when `labels` is provided):
+            Classification (or regression if config.num_labels==1) loss.
+        logits (`torch.FloatTensor` of shape `(batch_size, config.num_labels)`):
+            Classification (or regression if config.num_labels==1) scores (before SoftMax).
+        hidden_states (`tuple(torch.FloatTensor)`, *optional*, returned when `output_hidden_states=True` is passed or when `config.output_hidden_states=True`):
+            Tuple of `torch.FloatTensor` (one for the output of the embeddings + one for the output of each stage) of
+            shape `(batch_size, num_channels, height, width)`. Hidden-states (also called feature maps) of the model at
+            the output of each stage.
+        attentions (`tuple(torch.FloatTensor)`, *optional*, returned when `output_attentions=True` is passed or when `config.output_attentions=True`):
+            Tuple of `torch.FloatTensor` (one for each layer) of shape `(batch_size, num_heads, patch_size,
+            sequence_length)`.
```

Comment:
Not sure if patch_size, sequence_length is correct here.

Reply:
copied from `SemanticSegmentationModelOutput`

Reply:
Not sure the name is ideal either, `patch_size` should be replace by something that works for both ViT and ResNet

Reply:
well also `num_heads` and `sequence_length`, I am not sure how to proceed

Reply:
kindly pinging @sgugger

Reply:
Here, `(batch_size, num_heads, patch_size, sequence_length)` is not correct. For ViT for instance, the sequence length is equal to (config.image_size // config.patch_size)**2 + 1. So the attentions have shape (`batch_size, num_heads, sequence_length, sequence_length)`.

For Swin Transformer, this will be different (not sure what the sequence length is there). Swin has a different sequence length per layer.

Reply:
You are also defining an image classifier output here that includes attentions, but I guess most image classifiers won't have attention.

Reply:
I'd advocate to only keep `ImageClassifierOutput` in this PR, and for now use `BaseModelOutput` and `BaseModelOutputWithPooling` for the other classes. 

I'd like to work with placeholders rather than defining `BaseModelOutputWithNoAttentionAndWithPooling`.

cc @sgugger @FrancescoSaverioZuppichini

Reply:
I have no idea what your comment means, Niels. Could you elaborate? `BaseModelOutput` and `BaseModelOutputWithPooling` can't work for ResNet or ConvNext as they have the `attentions` field and those model do not.

Reply:
Yes that's what I meant.

Reply:
Resolved since we are going to add these outputs in this PR